node-notes-cli/app.js

```diff
@@ -0,0 +1,36 @@
+const fs = require('fs');
+const json = require('./data.json');
+const action = process.argv[2];
+
+if (action === 'read') {
+  for (let i = 1; i < `${json.nextId}`; i++) {
+    if (`${json.notes[i]}` !== 'undefined') {
+      console.log(`${i}: ${json.notes[i]}`);
+    }
+  }
+} else if (action === 'add') {
+  const note = process.argv[3];
+  json.notes[json.nextId++] = note;
+  fs.writeFile('data.json', `${JSON.stringify(json, null, 2)}`, err => {
+    if (err) throw err;
+  }
+  );
+} else if (action === 'update') {
+  const update = process.argv[3];
+  for (let i = 1; i < `${json.nextId}`; i++) {
+    if (parseInt(update) === i) {
+      json.notes[i] = process.argv[4];
+    }
+  }
```

Comment:
Ok, now we're back to this issue:

---

![image](https://user-images.githubusercontent.com/16143771/159073670-d8af7661-bc6b-4c3c-ad39-5c0be3620612.png)

---

The `update` operation needs to verify an id exists (meaning it hasn't, for example, been deleted) before updating it.